src/App.ts

```diff
@@ -6,84 +6,161 @@
   const app = express();
   app.set("port", config.CONTROLLER.PORT);
-  app.use(bodyParser.json());
-  app.use(bodyParser.urlencoded({ extended: false }));
-  setRestfulRoutes(app, config, verificaRPTPagoPaClient, attivaRPTPagoPaClient);
+  setRestfulRoutes(app, config, redisClient, pagoPAClient);
+  getSoapServer(
+    redisClient,
+    config.PAYMENT_ACTIVATION_STATUS_TIMEOUT as number
+  )(app);
+
   const server = http.createServer(app);
   server.listen(config.CONTROLLER.PORT);
+
   logger.info(
     `Server started at ${config.CONTROLLER.HOST}:${config.CONTROLLER.PORT}`
   );
   return server;
 }
 
 export function stopServer(server: http.Server): void {
-  logger.info("Stopping Proxy PagoPa Server...");
+  logger.info("Stopping Proxy PagoPA Server...");
   server.close();
 }
 
-// Set Restful WS routes
+/**
+ * Set RESTful WS endpoints
+ * @param {core.Express} app - Express server to set
+ * @param {Configuration} config - PagoPa proxy configuration
+ * @param {redis.RedisClient} redisClient - The redis client used to store information sent by PagoPA
+ * @param {PagamentiTelematiciPspNodoAsyncClient} pagoPAClient - PagoPa SOAP client to call verificaRPT and attivaRPT services
+ */
 function setRestfulRoutes(
   app: core.Express,
   config: Configuration,
-  verificaRPTPagoPaClient: pagoPaSoapClient.PagamentiTelematiciPspNodoAsyncClient,
-  attivaRPTPagoPaClient: pagoPaSoapClient.PagamentiTelematiciPspNodoAsyncClient
+  redisClient: redis.RedisClient,
+  pagoPAClient: pagoPASoapClient.PagamentiTelematiciPspNodoAsyncClient
 ): void {
+  const jsonParser = bodyParser.json();
+  const urlencodedParser = bodyParser.urlencoded({ extended: false });
   app.get(
-    config.CONTROLLER.ROUTES.RESTFUL.PAYMENTS_CHECK,
+    config.CONTROLLER.ROUTES.RESTFUL.PAYMENT_REQUESTS_GET,
+    urlencodedParser,
     (req: express.Request, res: express.Response) => {
       logger.info("Serving Payment Check Request (GET)...");
       toExpressHandler(
-        PaymentController.checkPaymentToPagoPa(
-          config.PAGOPA,
-          verificaRPTPagoPaClient
-        )
+        PaymentController.paymentRequestsGet(config.PAGOPA, pagoPAClient)
       )(req, res);
     }
   );
   app.post(
-    config.CONTROLLER.ROUTES.RESTFUL.PAYMENTS_ACTIVATION,
+    config.CONTROLLER.ROUTES.RESTFUL.PAYMENT_ACTIVATIONS_POST,
+    jsonParser,
     (req: express.Request, res: express.Response) => {
       logger.info("Serving Payment Activation Request (POST)...");
       toExpressHandler(
-        PaymentController.activatePaymentToPagoPa(
-          config.PAGOPA,
-          attivaRPTPagoPaClient
-        )
+        PaymentController.paymentActivationsPost(config.PAGOPA, pagoPAClient)
       )(req, res);
     }
   );
+  app.get(
+    config.CONTROLLER.ROUTES.RESTFUL.PAYMENT_ACTIVATIONS_GET,
+    urlencodedParser,
+    (req: express.Request, res: express.Response) => {
+      logger.info("Serving Payment Check Activation Request (GET)...");
```

Comment:
I sugget to replace all this calls to logger.info inside controllers with a [morgan middleware](https://github.com/expressjs/morgan), see:

https://github.com/teamdigitale/italia-backend/blob/e8fe4439f9e74a531b4e7d90a62bda23331dc894/src/app.ts#L116